extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVault.java

```diff
@@ -39,6 +40,10 @@ public HashicorpVault(@NotNull HashicorpVaultClient hashicorpVaultClient, @NotNu
     public @Nullable String resolveSecret(String key) {
         var result = hashicorpVaultClient.getSecretValue(key);
 
+        if (result.failed()) {
```

Comment:
Sorry, one other thing: there is no need to check for failure/success twice. Please combine the return statement with the failure check above.

Reply:
@jimmarino Now i check if the result failed and then return null + log or the result.